kubernetes/linux/main.sh

```diff
@@ -1073,19 +1073,32 @@ if [ -e "/opt/dcr_env_var" ]; then
       setGlobalEnvVar LOGS_AND_EVENTS_ONLY "${LOGS_AND_EVENTS_ONLY}"
 fi
 
-setGlobalEnvVar AZMON_RESOURCE_OPTIMIZATION_ENABLED "${AZMON_RESOURCE_OPTIMIZATION_ENABLED}"
+setGlobalEnvVar ENABLE_CUSTOM_METRICS "${ENABLE_CUSTOM_METRICS}"
+if [ "${ENABLE_CUSTOM_METRICS}" == "true" ]; then
+      setGlobalEnvVar AZMON_RESOURCE_OPTIMIZATION_ENABLED "false"
+      export AZMON_RESOURCE_OPTIMIZATION_ENABLED="false"
+else
+      setGlobalEnvVar AZMON_RESOURCE_OPTIMIZATION_ENABLED "${AZMON_RESOURCE_OPTIMIZATION_ENABLED}"
+fi
+
 if [ "$AZMON_RESOURCE_OPTIMIZATION_ENABLED" != "true" ]; then
       # no dependency on fluentd for prometheus side car container
       if [ "${CONTAINER_TYPE}" != "PrometheusSidecar" ] && [ "${GENEVA_LOGS_INTEGRATION_SERVICE_MODE}" != "true" ]; then
             if [ ! -e "/etc/config/kube.conf" ]; then
                   if [ "$LOGS_AND_EVENTS_ONLY" != "true" ]; then
                         echo "*** starting fluentd v1 in daemonset"
+                        if [ "${ENABLE_CUSTOM_METRICS}" == "true" ]; then
+                              mv /etc/fluent/container-cm.conf /etc/fluent/container.conf
+                        fi
                         fluentd -c /etc/fluent/container.conf -o /var/opt/microsoft/docker-cimprov/log/fluentd.log --log-rotate-age 5 --log-rotate-size 20971520 &
                   else
                         echo "Skipping fluentd since LOGS_AND_EVENTS_ONLY is set to true"
                   fi
             else
                   echo "*** starting fluentd v1 in replicaset"
+                  if [ "${ENABLE_CUSTOM_METRICS}" == "true" ]; then
+                        mv /etc/fluent/kube-cm.conf /etc/fluent/kube.conf
```

Comment:
I think we should test all the different windows cases as well with this PR to make sure nothing is breaking: 
- Legacy and MSI
- MSI with and without AZMON_WINDOWS_FLUENT_BIT_DISABLED
- Geneva (legacy and MSI)